tasks/columns.py

```diff
@@ -0,0 +1,60 @@
+import json
+from collections import OrderedDict
+
+
+CONDITIONS = "conditions"
+EXCEPTIONS = "exceptions"
+
+
+class ColumnsDeclarations:
+    def __init__(self, JSONFile):
+        with open(JSONFile) as file:
+            self._columns = json.load(file)
+
+    def _find_column(self, colid):
+        return self._columns[colid]
+
+    def _get_requirements(self, colid, requirement):
+        try:
+            return self._find_column(colid)[requirement]
+        except:
+            return None
+
+    def _is_column_filtered(self, colid, parameters):
+        params = json.loads(parameters)
+
+        conditions = self._get_requirements(colid, CONDITIONS)
+        if conditions is not None:
+            if not self._check_requirements(params, conditions):
+                return False
+
+        exceptions = self._get_requirements(colid, EXCEPTIONS)
+        if exceptions is not None:
+            if self._check_requirements(params, exceptions):
+                return False
+
+        return True
+
+    def _check_requirements(self, parameters, requirements):
+        if not requirements:
+            return True
+
+        for requirement in requirements:
+            check = True
+            for id, value in parameters.iteritems():
+                try:
+                    if requirement[id] != value:
+                        check = False
+                except:
+                    pass
+            if check:
+                return True
```

Comment:
Some ideas to simplify:

Just use `if id in requirements and requirements[id] != value` instead of the try/catch.

I think you can just do `return False` in the requirement, and `return True` after the loop, and remove the `check` variable entirely.

Bonus trick, have you heard about `for else`? https://docs.python.org/2/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops

Also, avoid using `id` since it is a Python built-in.

Reply:
:heart: the `for else`